### ISSUE TITLE
POST to Chassis.Reset for invalid chassis ID returns 200 OK status code

### ISSUE THREAD

**Issue by FarahRasheed1** [openbmc/bmcweb#289]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use OpenBMC
- [X] This is not a bug in an OpenBMC fork or a bug in code still under code review.
- [X] This is not a request for a new feature.

## Problem ##

Sending a POST request to reset an invalid chassis that does not exist returns 200 OK. I reproduced this on QEMU.

It looks like the service is not validating the chassis ID correctly for `Chassis.Reset` action , leading it to respond with 200 OK even when the chassis doesn't exist.

### Curl command: ###

```
curl -v -k -u root:0penBmc -X POST https://localhost:2443/redfish/v1/Chassis/random_chassis_that_does_not_exist/Actions/Chassis.Reset \
-H "Content-Type: application/json" \
-d '{
    "ResetType": "PowerCycle"
}'
```

### Response: ###

```
{
  "@Message.ExtendedInfo": [
    {
      "@odata.type": "#Message.v1_1_1.Message",
      "Message": "The request completed successfully.",
      "MessageArgs": [],
      "MessageId": "Base.1.13.0.Success",
      "MessageSeverity": "OK",
      "Resolution": "None"
    }
  ]
}
```

### Logs ###

`Sep 17 14:48:32 romulus phosphor-chassis-state-manager[291]: Change to Chassis Requested Power State: xyz.openbmc_project.State.Chassis.Transition.PowerCycle`

### Version

```Text
ID=openbmc-openpower
NAME="Phosphor OpenBMC (Phosphor OpenBMC Project Reference Distro)"
VERSION="2.14.0"
VERSION_ID=2.14.0
VERSION_CODENAME="langdale"
PRETTY_NAME="Phosphor OpenBMC (Phosphor OpenBMC Project Reference Distro) 2.14.0"
BUILD_ID="20240724195151"
OPENBMC_TARGET_MACHINE="romulus"
EXTENDED_VERSION="2.14.0"
```

**Comment by edtanous** [comment]
Pretty sure this is because all chassis resets are routed to one dbus path, not a per-chassis path.  I think @wltu had some patches a while back trying to clean this behavior up, but I don't know where they landed.

Thanks for the bug report!

**Comment by williamspatrick** [comment]
I assume this is also being worked by @mox669 who is doing the multi-host implementation.

**Comment by FarahRasheed1** [comment]
Yup, I had found https://gerrit.openbmc.org/c/openbmc/bmcweb/+/45396 which at a first look might fix this bug, but figured since there were no open issues, I may as well open it to keep track of it.

**Comment by gtmills** [comment]
Just like the PATCH, we should verify the Chassis exists on D-Bus on the POST 
https://github.com/openbmc/bmcweb/blob/master/redfish-core/lib/chassis.hpp#L859

**Comment by gtmills** [comment]
Merged at https://gerrit.openbmc.org/c/openbmc/bmcweb/+/90193
